merino/jobs/navigational_suggestions/domain_metadata_extractor.py

```diff
@@ -236,56 +282,63 @@ def _extract_favicons(self, scraped_url: str) -> list[dict[str, Any]]:
 
         return favicons
 
-    def _get_best_favicon(self, favicons: list[dict[str, Any]], min_width: int) -> str:
-        """Return the favicon with the highest resolution that satisfies the minimum width
-        criteria from a list of favicons.
-        """
+    async def _get_best_favicon(self, favicons: list[dict[str, Any]], min_width: int) -> str:
+        """Asynchronous method to find the best favicon"""
+        urls = [self._fix_url(favicon["href"]) for favicon in favicons]
+        masked_svg_indices = [i for i, favicon in enumerate(favicons) if "mask" in favicon]
+
         best_favicon_url = ""
         best_favicon_width = 0
-        for favicon in favicons:
-            url = self._fix_url(favicon["href"])
-            width = None
-            favicon_image: Image | None = self.favicon_downloader.download_favicon(url)
-            if favicon_image is None:
-                continue
-
-            if "image/" not in favicon_image.content_type:
-                # If favicon mime type is not "image" then skip it
-                continue
-
-            # If favicon is an SVG, then return this as the best favicon because SVG favicons
-            # are scalable, can be printed with high quality at any resolution and SVG
-            # graphics do NOT lose any quality if they are zoomed or resized.
-            if favicon_image.content_type == "image/svg+xml":
-                # Firefox doesn't support masked favicons yet. Return if not masked.
-                if "mask" not in favicon:
-                    return url
-                else:
-                    logger.info(f"Masked SVG favicon {favicon} found; skipping it")
+
+        try:
+            # Process favicons in chunks to limit concurrent connections
+            chunk_size = 20
+            all_favicon_images = []
+
+            for i in range(0, len(urls), chunk_size):
+                chunk_urls = urls[i : i + chunk_size]
```

Comment:
FYI: `itertools` comes with a function `batched` for this if you like. 
```suggestion
            for chunk_urls in itertools.batched(urls, chunk_size):
```